LU afolu variables

flohump opened the pull request:
This revision of reporting variables is part of a larger set of changes related to AFOLU reporting variables that have been discussed and agreed with representative of all modeling teams in @IAMconsortium/common-definitions-land 
The revision is based on https://1drv.ms/x/s!AlN2Seu8OCz8hLQ8Wh_ysgHyDSg4QA
For consistency with naming conventions some variables have been named and defined differently.
The purpose of tier, domain and subdomain is to allow for automated checks of reported variables in specific domains.
Due to the use of tag yaml files some tier levels could not be set as discussed. 
@danielhuppmann Is it possible to use tier levesl in tag yaml files?

flohump commented:
@danielhuppmann Let me know if you need more information from me for the review of the PR.

danielhuppmann commented on definitions/variable/afolu/food_system.yaml:
```diff
@@ -0,0 +1,92 @@
+- Food Availability [per capita]:
```
Please name this file as `food-system.yaml` for consistency. And if you think it's necessary to move delete the file `food.yaml` and move "Food Demand" somewhere else, why not here?

flohump replied:
Done.

danielhuppmann commented on definitions/variable/afolu/tag_forestry_types.yaml:
```diff
@@ -0,0 +1,24 @@
+- Raw products:
```
Please use capitalized tag names for consistency.
```suggestion
- Raw Products:
```

danielhuppmann commented on definitions/variable/afolu/tag_forestry_types.yaml:
```diff
@@ -0,0 +1,24 @@
+    - Other wood products:
+        description: Other wood products
+        unit: million m3/yr
+    - Mechanical pulp:
```
Does this need a super-category "Pulp"?

flohump replied:
No. Not needed from my perspective.

danielhuppmann commented on definitions/variable/afolu/agriculture.yaml:
```diff
@@ -1,156 +1,309 @@
+    tier: 1
+    domain: AFOLU
+    subdomain: Agricultural Prices
+- Price|Agriculture|Food Products|Crops|Corn [Index]:
```
Why is "Corn" not included in the list of crops for the "Agricultural Demand" and other variables?

flohump replied:
I renamed it to "Maize" for consistency. "Corn" is more commonly used in North America.

danielhuppmann reviewed (changes requested):
This PR has too many changes to review properly, so I added some comments but no guarantee on completeness.

Also, I'm confused why you move around existing variables and tag-lists.

And As mentioned in another PR adding domains and subdomains should be first agreed across the entire repository, with useful guidance.

flohump commented on definitions/variable/afolu/tag_forestry_types.yaml:
```diff
@@ -0,0 +1,24 @@
+        description: Industrial Roundwood
+    - Wood fuel:
+        description: Wood fuel
+- Semi-finished products:
```
Thanks for flagging. I now put all tag lists into single files including capitalized names.

flohump commented on definitions/variable/afolu/land-cover.yaml:
```diff
@@ -1,82 +1,138 @@
+    subdomain: Land Cover
+
+- Land Cover|Other Natural:
+    description: Non-forest natural land - 
+      sum of primary non-forest, restored and recovered natural land
```
Are hyphens generally not allowed?

flohump replied:
done

flohump commented on definitions/variable/afolu/land-cover-change.yaml:
```diff
@@ -0,0 +1,91 @@
+    tier: 3
+    domain: AFOLU
+    subdomain: Land Cover
+- Forest Area Change|Gross Change|Deforestation:
```
revised because not needed in variable name. Thanks for the suggestion.

flohump commented on definitions/variable/afolu/land-cover-change.yaml:
```diff
@@ -0,0 +1,91 @@
+    tier: 3
+    domain: AFOLU
+    subdomain: Land Cover
+- Forest Area Change|Gross Change|Forest Expansion:
```
Yes. revised.

flohump commented:
> This PR has too many changes to review properly, so I added some comments but no guarantee on completeness.

Thanks for the review! And sorry that the PR is still so large. But I could not think of a clever way to split the PR without making my life too complicated. 

> Also, I'm confused why you move around existing variables and tag-lists.

I moved all variables that are directly related to agriculture to `agriculture.yaml`, like demand, production, trade, yields and prices. 
Same for forestry in `forestry.yaml`
I hope this makes sense. 
 
> And As mentioned in another PR adding domains and subdomains should be first agreed across the entire repository, with useful guidance.

I understand your concerns. The main idea behind this is to allow for machine-readalbe checks of tier-1, tier-2 and tier-3 reportings in a specific domain, Land / AFOLU in this case.
Given the time constraints, I will remove these attributes for now.

danielhuppmann commented on definitions/variable/afolu/forestry.yaml:
```diff
@@ -1,19 +1,82 @@
+    tier: 2
+
+# Forestry Demand, Production and Trade of Semi-finished Products
+- Forestry Demand Semi-finished|{Semi-Finished Products}:
```
```suggestion
- Forestry Demand Semi-Finished|{Semi-Finished Products}:
```

danielhuppmann commented on definitions/variable/afolu/forestry.yaml:
```diff
@@ -1,19 +1,82 @@
+    tier: 2
+
+# Forestry Harvested Area
+- Harvested area|Forestry:
```
```suggestion
- Harvested Area|Forestry:
```

danielhuppmann commented on definitions/variable/afolu/land-cover-change.yaml:
```diff
@@ -0,0 +1,65 @@
+- Forest Area Change:
+    description: Annual net change in total forest area between last period and this 
+      period (difference divided by period length)
```
For consistency with other variables (see #177), I suggest to split the description and the notes
```suggestion
    description: Annual net change in total forest area
    notes: This variable should be computed as yearly average change between the previous
      and the current reported time step.
```

danielhuppmann commented on definitions/variable/afolu/land-cover-change.yaml:
```diff
@@ -0,0 +1,65 @@
+      (difference divided by period length)
+    unit: million ha/yr
+    tier: 3
+- Forest Area Change|Forest Expansion:
```
Is this really something different than Afforestation? Given that this is in a variable referring explicitly to the area change...

flohump replied:
Yes, it is. Forest expansion also includes natural regeneration in secondary forest

danielhuppmann commented on definitions/variable/afolu/land-cover-change.yaml:
```diff
@@ -0,0 +1,65 @@
+      period (difference divided by period length)
+    unit: million ha/yr
+    tier: 3
+- Forest Area Change|Forest Expansion|Planted|Plantation|Re-Afforestation:
```
It's not clear to me what Re-Afforestation is, in particular if it's a companion to "timber plantations", but then described as carbon plantation...

flohump replied:
revised definition: Annual gross expansion of carbon plantations; Re-Afforestation with monoculture and/or non-native species

danielhuppmann commented on definitions/variable/afolu/forestry.yaml:
```diff
@@ -1,19 +1,82 @@
-- Forestry Demand|Roundwood|Wood Fuel:
-    description: Wood fuel demand
+    tier: 2
+- Forestry Demand Roundwood|Industrial Roundwood|Material use:
```
```suggestion
- Forestry Demand Roundwood|Industrial Roundwood|Material Use:
```

danielhuppmann commented on definitions/variable/afolu/forestry.yaml:
```diff
@@ -1,19 +1,82 @@
-    description: Wood fuel demand
+    tier: 2
+- Forestry Demand Roundwood|Industrial Roundwood|Material use:
+    description: Industrial roundwood demand - used for material
```
Please make the description of bit more helpful.

flohump replied:
Done: Industrial roundwood demand - used as material for wood products

danielhuppmann commented on definitions/variable/afolu/forestry.yaml:
```diff
@@ -1,19 +1,82 @@
+    description: Industrial roundwood demand - used for material
     unit: million m3/yr
+    tier: 2
+- Forestry Demand|Roundwood|Industrial Roundwood|Energy use:
```
```suggestion
- Forestry Demand|Roundwood|Industrial Roundwood|Energy Use:
```

danielhuppmann commented on definitions/variable/afolu/forestry.yaml:
```diff
@@ -1,19 +1,82 @@
+    tier: 3
+
+# Forestry Yield
+- Yield|Forestry|Timber production:
```
```suggestion
- Yield|Forestry|Timber Production:
```

danielhuppmann reviewed (changes requested):
This PR adds ore than 100 new variables across a dozen categories and removes 20 other variables, moves existing variables to new files, changes the order of variables within files... This makes it very difficult to review.

Also, this PR still has `domain` and `subdomain` attributes (particularly in files unrelated to any of the major changes in this PR).

Final major comment - this PR is inconsistent in the structure of variables related to roundwood.
- Forestry Demand Roundwood
- Forestry Demand Roundwood|Industrial Roundwood|Material Use

but
- Forestry Demand|Roundwood|Industrial Roundwood|Energy Use

So here we have Roundwood as a sub-category of demand.

I think having Roundwood as a sub-category, in parallel to "Semi-Finished Products", makes a lot more sense, but this needs to be implemented consistency in Forestry Production, Forestry Trade, and Prices|Forestry..

flohump commented:
> This PR adds ore than 100 new variables across a dozen categories and removes 20 other variables, moves existing variables to new files, changes the order of variables within files... This makes it very difficult to review.

I'm sorry about that. But I also see no other meaningful way for splitting.
As indicated before, the overall revision of the LU / AFOLU variables has been agreed in a virtual meeting with LU representatives of different IAM teams. 
@jonathandoelman Could you have another look at the PR for cross-checking?

> Also, this PR still has `domain` and `subdomain` attributes (particularly in files unrelated to any of the major changes in this PR).

Indeed, I missed these. Removed now. 

> Final major comment - this PR is inconsistent in the structure of variables related to roundwood.
> 
> * Forestry Demand Roundwood
> * Forestry Demand Roundwood|Industrial Roundwood|Material Use
> 
> but
> 
> * Forestry Demand|Roundwood|Industrial Roundwood|Energy Use
> 
> So here we have Roundwood as a sub-category of demand.
> 
> I think having Roundwood as a sub-category, in parallel to "Semi-Finished Products", makes a lot more sense, but this needs to be implemented consistency in Forestry Production, Forestry Trade, and Prices|Forestry..

Good point. My initial suggestion was based on idea of sub-categories that add up to the total. I agree that here your suggestion makes more sense. 
I revised all related variables.

jonathandoelman commented on definitions/variable/afolu/tag_forestry-raw-products.yaml:
in previous rounds I think we also distinguished between pulp and sawlogs, is that dropped on purpose? no high priority though

flohump replied:
That's covered in the Semi-Finished wood products.

jonathandoelman commented on definitions/variable/afolu/tag_forestry-semi-finished.yaml:
@flohump do you know why we in some cases use m3 and in others tDM? wouldnt it be easier to more simple to always use tDM? or is m3 closer to the original unit of FAO

flohump replied:
This is based on a suggestion from the GLOBIOM team. I a bilateral E-Mail conversion they confirmed that in line with FAO pulp should have a different unit (million tDM/yr) than all other forestry products (million m3/yr).

danielhuppmann commented:
Thanks @flohump for the revisions, just one question (similar to https://github.com/IAMconsortium/common-definitions/pull/172#issuecomment-2440622435) - what is "Re-Afforestation"? In one of the descriptions, it is written as "Re/afforestation", is it basically just a short-hand notation for "Re- and Afforestation"?

jonathandoelman commented on definitions/variable/afolu/agriculture.yaml:
Why did we change from Trade|Agricultural to Agricultural Trade? I cant find the equivalent for energy, is it done similarly there now? I dont have a clear preference, but in previous versions in was usually first trade and then the category, right?

jonathandoelman replied:
I agree with Tomoko that we should make Yields a tier 1 variable

jonathandoelman replied:
@flohump I think its odd that fertilizer phosphorus is here, and all nitrogen stuff is part of SDGs. we could consider making an environment category and put biodiversity, nutrients and water there. also nice to split to socio-economics and environmental indicators of the SDGs. but this is a nice-to-have.

jonathandoelman replied:
for prices I think it is inconsistent that we have a different set of crop categories (corn, rice, soy, wheat) compared to the agricultural production/demand. wouldnt it make sense to make those consistent? so change prices also to cereals, oil, other and sugar crops?

jonathandoelman replied:
the weights for yield and prices do not seem to be specific enough, is that an error?

flohump replied:
> Why did we change from Trade|Agricultural to Agricultural Trade? I cant find the equivalent for energy, is it done similarly there now? I dont have a clear preference, but in previous versions in was usually first trade and then the category, right?

We didn't have `Agricultural Trade (million t DM/yr)` at all before. We could rename it to `Trade|Agriculture`, which would be more in line with Energy e.g. `Trade|Primary Energy|Biomass (EJ/yr) and (billion USD_2010/yr)`. However, the units are different and thus summing is not possible. Moreover, `Trade|Agriculture` would differ in the naming logic from `Agricultural Demand (million t DM/yr)` and `Agricultural Production (million t DM/yr)`
@danielhuppmann What do you think?

flohump replied:
> I agree with Tomoko that we should make Yields a tier 1 variable

We agreed to have Tier-1 for Energy and Cereal crops yields, and 2 for all other yields. 
However, with the tag list structure I don't know how to implement this. 
Therefore, I used Tier-2 for all crop yields.
@danielhuppmann Is is possible to have different tier levels in the yaml tag files?

flohump replied:
> @flohump I think its odd that fertilizer phosphorus is here, and all nitrogen stuff is part of SDGs. we could consider making an environment category and put biodiversity, nutrients and water there. also nice to split to socio-economics and environmental indicators of the SDGs. but this is a nice-to-have.

I fully agree. But I was hesitant to move nitrogen, biodiversity and water out of the `sdg-indicators` folder. 
If I'm not mistaken, this collection comes from the SHAPE project. 
However, the intention of the `sdg-indicators` folder is not fully clear to me. @danielhuppmann Is the goal to have all SDG indicators in this folder? But this would conflict with many other categorisations, e.g. primary forest in `afolu/land-cover.yaml` Or is `sdg-indicators` just a big collection of indicators that should actually be moved somewhere else if possible?

@jonathandoelman Do you suggest to create an `environment` folder with files for biodiversity, nutrients and water? To me this sounds meaningful. Would this replace the `water` folder?

flohump replied:
> for prices I think it is inconsistent that we have a different set of crop categories (corn, rice, soy, wheat) compared to the agricultural production/demand. wouldnt it make sense to make those consistent? so change prices also to cereals, oil, other and sugar crops?

Aggregation of prices is often not trivial. Therefore, we have prices for individual crops (like maize) and not crop groups (like cereals).

flohump replied:
> the weights for yield and prices do not seem to be specific enough, is that an error?

For yields, it's a pragmatic solution because I don't know if it is possible / safe to use `weight` in tag lists, considering that the tag lists are also used for non-intensive variables. @danielhuppmann 
Anyways, I think `Land Cover|Cropland` is a good proxy. 
For prices, we don't have individual products like maize or wheat in Agricultural Production.

jonathandoelman replied:
> > I agree with Tomoko that we should make Yields a tier 1 variable
> 
> We agreed to have Tier-1 for Energy and Cereal crops yields, and 2 for all other yields. However, with the tag list structure I don't know how to implement this. Therefore, I used Tier-2 for all crop yields. @danielhuppmann Is is possible to have different tier levels in the yaml tag files?

ah that sounds sensible indeed, waiting for @danielhuppmann if this is possible.

danielhuppmann replied:
We are working on a feature to support different tiers across tag-list. For now, please leave all crop yields at 2 so that we can merge this PR and let's revisit after IAMC.

jonathandoelman replied:
> > @flohump I think its odd that fertilizer phosphorus is here, and all nitrogen stuff is part of SDGs. we could consider making an environment category and put biodiversity, nutrients and water there. also nice to split to socio-economics and environmental indicators of the SDGs. but this is a nice-to-have.
> 
> I fully agree. But I was hesitant to move nitrogen, biodiversity and water out of the `sdg-indicators` folder. If I'm not mistaken, this collection comes from the SHAPE project. However, the intention of the `sdg-indicators` folder is not fully clear to me. @danielhuppmann Is the goal to have all SDG indicators in this folder? But this would conflict with many other categorisations, e.g. primary forest in `afolu/land-cover.yaml` Or is `sdg-indicators` just a big collection of indicators that should actually be moved somewhere else if possible?
> 
> @jonathandoelman Do you suggest to create an `environment` folder with files for biodiversity, nutrients and water? To me this sounds meaningful. Would this replace the `water` folder?

I think the sdg.yml is a weird mix so I would like to split that up. more detail is better right? so maybe we can propose to rename the folder 'water' to 'environment', and then keep the water.yml there, and add seperate yml-files for nutrients and biodiversity. I think thats a lot more clear.

danielhuppmann replied:
Restructuring existing variables is ok, but please do that as a follow-up PR. This discussion is going down a rabbit hole.

bs538 replied:
Yes, indeed the sdg.yaml is a bucket list of additional SDG indicators from various categories (historically that was variables not included in the NAVIGATE template). If we now move to a more detailed land indicator set, I'm generally all fine with moving relevant variables there. My only request would be to ensure that they are tagged to a certain SDG, so we can find them conveniently for SDG reporting. 

But agree with Daniel, please make that a separate PR where we can discuss in more detail if needed.

jonathandoelman commented on definitions/variable/sdg-indicators/sdg.yaml:
whats the difference between synthetic fertilizer and 'industrial and intentional biological fixation'?

flohump replied:
I guess `industrial and intentional biological fixation` is the sum of synthetic fertilizer and Biological Fixation. But not entirely sure. For consistency and simplifcation, I suggest to condense `Nitrogen|Cropland Budget|Inputs|xxx` to `Fertilizer Use|Nitrogen|Organic` and `Fertilizer Use|Nitrogen|Synthetic`.

jonathandoelman replied:
yes I think that is better, few models represent the nutrient budget in detail, and also for IMAGE we wont produce those results for scenariomip. the fertilizer use may be easier to report.

jonathandoelman reviewed (commented):
Hi @flohump and @danielhuppmann; I went through the whole list, it looks very good. I added a number of comments/small issues. maybe we can still improve some things, Florian. but if we are in a hurry I didnt find any major issues so I am also happy to approve and put a few suggestions on an issue list.
Jonathan

flohump commented:
> Hi @flohump and @danielhuppmann; I went through the whole list, it looks very good. I added a number of comments/small issues. maybe we can still improve some things, Florian. but if we are in a hurry I didnt find any major issues so I am also happy to approve and put a few suggestions on an issue list. Jonathan

Thanks for the in-depth review! I have a couple of follow-up questions to you and @danielhuppmann. Please see my comments.

flohump commented:
> Thanks @flohump for the revisions, just one question (similar to [#172 (comment)](https://github.com/IAMconsortium/common-definitions/pull/172#issuecomment-2440622435)) - what is "Re-Afforestation"? In one of the descriptions, it is written as "Re/afforestation", is it basically just a short-hand notation for "Re- and Afforestation"?

Yes, it is just a short-hand for "Re- and Afforestation". Do you suggest to use this instead of "Re-Afforestation" in all occurrences? I would be fine with this.

tmkhsgw reviewed (commented):
I went through the files.

danielhuppmann commented:
> > Thanks @flohump for the revisions, just one question (similar to [#172 (comment)](https://github.com/IAMconsortium/common-definitions/pull/172#issuecomment-2440622435)) - what is "Re-Afforestation"? In one of the descriptions, it is written as "Re/afforestation", is it basically just a short-hand notation for "Re- and Afforestation"?
> 
> Yes, it is just a short-hand for "Re- and Afforestation". Do you suggest to use this instead of "Re-Afforestation" in all occurrences? I would be fine with this.

Well, my initial thought was that this was some kind of restriction of afforestation to areas that previously had forest cover - definitely not obvious that this is an either-or. How about using either of these two instead?
- Re/Aforrestation
- Re- and Afforestation
in the variable name and being more clear in the descriptions, like "both reforestation and afforestation..."

jonathandoelman commented:
> > > Thanks @flohump for the revisions, just one question (similar to [#172 (comment)](https://github.com/IAMconsortium/common-definitions/pull/172#issuecomment-2440622435)) - what is "Re-Afforestation"? In one of the descriptions, it is written as "Re/afforestation", is it basically just a short-hand notation for "Re- and Afforestation"?
> > 
> > 
> > Yes, it is just a short-hand for "Re- and Afforestation". Do you suggest to use this instead of "Re-Afforestation" in all occurrences? I would be fine with this.
> 
> Well, my initial thought was that this was some kind of restriction of afforestation to areas that previously had forest cover - definitely not obvious that this is an either-or. How about using either of these two instead?
> 
> * Re/Aforrestation
> * Re- and Afforestation
>   in the variable name and being more clear in the descriptions, like "both reforestation and afforestation..."

I prefer Re/Afforestation :)

danielhuppmann commented on definitions/variable/sdg-indicators/sdg.yaml:
Please add that as a follow-up PR.

flohump replied:
Fertilizer Use|Phosphorus exists in agriculture.yaml. 
Fertilizer Use|Nitrogen is the same as Nitrogen|Cropland Budget|Inputs.
But as requested by @danielhuppmann I will do this a as follow-up PR.

danielhuppmann commented on definitions/variable/afolu/food-system.yaml:
Additional attributes are not a problem.

flohump replied:
>Weight should be Land Cover|Cropland|{Crop Types} instead of Land Cover|Cropland

I think this does not work technically because we would need to add `weight` as attribute in the tag file, while the tag itself is defining the variable.
E.g. 
- Crop Types:
    - Cereals:
        description: cereals (maize, rice, temperate cereals and tropical cereals)
        weight: ???

flohump replied:
I will leave the weight as is for now.

flohump commented on definitions/variable/afolu/land-management.yaml:
Ok. Done.

flohump commented on definitions/variable/afolu/land-cover-change.yaml:
For better consistency, I used slightly different variable names than in our AFOLU Excel spreadsheet in some cases. 
I stopped updating our AFOLU Excel spreadsheet. 
Teams will get a new Excel spreadsheet from common-definitons. 

Forest Area Change|Forest Expansion|Planted|Natural|Other = Forest Area Change|Forest Expansion|Planted|Other Planted Forest

flohump commented on definitions/variable/afolu/land-cover.yaml:
I labeled it "Land Cover|Other Natural|Recovered" by intention because "Land Cover|Other Natural|Abandoned" is confusing. It's not abandoned other land but abandoned ag land that is now recovering. I hope this makes sense. 

Land Cover|Other Natural: tier changed to 1. 
Land Cover|Cropland|{Crop Types}: tier is already 2
Land Cover|Cropland|{Crop Types}|Rainfed: now added
Land Cover|Cropland|{Crop Types}|Irrigated: now added

flohump commented on definitions/variable/afolu/forestry.yaml:
Weight for Yield|Forestry|Timber Production added. Thanks!
The AFOLU Excel file is no longer updated.

flohump commented:
> > > Thanks @flohump for the revisions, just one question (similar to [#172 (comment)](https://github.com/IAMconsortium/common-definitions/pull/172#issuecomment-2440622435)) - what is "Re-Afforestation"? In one of the descriptions, it is written as "Re/afforestation", is it basically just a short-hand notation for "Re- and Afforestation"?
> > 
> > 
> > Yes, it is just a short-hand for "Re- and Afforestation". Do you suggest to use this instead of "Re-Afforestation" in all occurrences? I would be fine with this.
> 
> Well, my initial thought was that this was some kind of restriction of afforestation to areas that previously had forest cover - definitely not obvious that this is an either-or. How about using either of these two instead?
> 
> * Re/Aforrestation
> * Re- and Afforestation
>   in the variable name and being more clear in the descriptions, like "both reforestation and afforestation..."

I changed to "Re/Afforestation" in the variable name and added more detail in the definition.

There are different definitions of Re/Afforestation.
a) previous state of area: was a forest before = reforestation; was not a forest before = afforestation
b) type of new forest established: natural forest = reforestation; monoculture/non-native species = afforestation.
a) is often ambiguous because the number of years until something should be counted as afforestation is not well defined. 
b) is more informative in particular for biodiversity aspects. 
Therefore, we use b) here.

flohump commented:
> I went through the files. Some are just notes about e.g. inconsistency with the excel var list. If you feel unnecessary here, please skip it.

Thanks for your detailed review. I hope that I have adressed all your points.

flohump commented:
@danielhuppmann It seems that we have converged on a consolidated version. Feel free to merge the PR or let me know in case further refinements are needed.

danielhuppmann commented:
Thank you all, this looks very nice! I have a few more minor comments for clean-up, but I will implement them as a follow-up PR.

One larger question: I'm a bit worried about adding too many top-level categories, and it seems that "Agricultural Trade" and "Forestry Trade" did not previously exist - so could we change that to "Trade|Agriculture|..." and "Trade|Forestry|..." for consistency with "Trade|Primary Energy|...", "Trade|Goods", etc.

flohump commented:
> One larger question: I'm a bit worried about adding too many top-level categories, and it seems that "Agricultural Trade" and "Forestry Trade" did not previously exist - so could we change that to "Trade|Agriculture|..." and "Trade|Forestry|..." for consistency with "Trade|Primary Energy|...", "Trade|Goods", etc.

I'm fine with that. I just pushed the changes.

danielhuppmann reviewed (approved):
Looks good to me, thanks!
